spring-components/rdf4j-spring/src/main/java/org/eclipse/rdf4j/spring/dao/support/RelationMapBuilder.java

```diff
@@ -146,14 +146,17 @@ private IRI getRelationValueOrNothing(BindingSet b) {
 
 	private TupleQueryEvaluationBuilder makeTupleQueryBuilder() {
 		return rdf4JTemplate
-				.tupleQuery(
-						Queries.SELECT(getProjection())
-								.where(getWhereClause())
-								.distinct()
-								.getQueryString())
+				.tupleQuery(makeQueryString())
 				.withBindings(bindingsBuilder.build());
 	}
 
+	String makeQueryString() {
```

Comment:
Can this be private?

Reply:
It is package-protected so it can be used in the RelationMapBuilderTests